parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -228,13 +235,16 @@ public ColumnWriteStore newColumnWriteStore(
           pageStore,
           pageSize,
           dictionaryPageSizeThreshold,
-          enableDictionary, writerVersion);
+          enableDictionary,
+          initialRowCountForSizeCheck,
+          estimateNextSizeCheck,
+          writerVersion);
     case PARQUET_2_0:
       return new ColumnWriteStoreV2(
           schema,
           pageStore,
           pageSize,
-          new ParquetProperties(dictionaryPageSizeThreshold, writerVersion, enableDictionary));
+          new ParquetProperties(dictionaryPageSizeThreshold, writerVersion, enableDictionary, initialRowCountForSizeCheck, estimateNextSizeCheck));
```

Comment:
I'll take a look at the v2 path.  It's changed from the v1 path and I'm not sure if the logic is the same.